### DISCUSSION TITLE
是否考虑直接抓包取得数据而不是ocr

### DISCUSSION THREAD

**Discussion by ghost** [NateScarlet/auto-derby#169] [General]
虽然赛马娘和服务器的数据包是加密过的，但是可以在本地解密后读取包，已经有人实现了hook libnative.dll来读取包数据https://github.com/CNA-Bld/EXNOA-CarrotJuicer
看样子基本上育成的所有事件都要和服务器通讯，理论上可以通过读取包来更加精确而全面地实现自动育成

**Comment by fatinghenji** [comment]
抓包比ocr更危险吧……
而且抓包能得到的数据ocr都能得到？

**Reply by ghost** [reply]
抓包能得到挺多ocr很难读或者读不出来的抽象东西 像体力支援卡exp具体值，以及竞技场可以在对手选择界面提前得知对手的马娘属性和评价点，避免平均评价点高的对手（高Class似乎不用关心这事）
确实抓包涉及到改动游戏本体，不过cy如果不查的话还算是个挺好的选择

**Reply by NateScarlet** [reply]
> 确实抓包涉及到改动游戏本体，不过cy如果不查的话还算是个挺好的选择

改客户端用来PVP怎么可能不查 除非你改了还是打不过没改的 那改个什么劲

**Reply by NateScarlet** [reply]
而且竞技场避免高评价对手=避免获胜后的高评价 我不懂为啥要这么做

**Reply by fatinghenji** [reply]
他说的避免高评分，应该指的是自己跑不过的“高评分”

**Reply by NateScarlet** [reply]
这游戏随机性很强 不跑是不知道跑不跑得过的

**Comment by NateScarlet** [comment]
不考虑

辅助脚本仅模拟用户操作 #100 我不玩PVP所以这对我没吸引力徒增风险

[而且有人似乎已经用插件实现了](https://github.com/CNA-Bld/auto-derby/tree/c97dfa8d0b2fb3d56541285900e912ecf5adef7b) 不过服务端没公开

**Reply by NateScarlet** [reply]
然后我发现用插件实现的和你发的抓包项目都是同一个人的
